classes/plugins/PubIdPlugin.inc.php

```diff
@@ -136,17 +136,21 @@ function getPubId($pubObject) {
 		$submissionFile = ($pubObjectType == 'SubmissionFile' ? $pubObject : null);
 
 		// Get the context id.
+		$submissionDao = Application::getSubmissionDAO();
 		if (in_array($pubObjectType, array('Issue', 'Submission'))) {
 			$contextId = $pubObject->getJournalId();
 		} else {
 			// Retrieve the submission.
 			assert(is_a($pubObject, 'Representation') || is_a($pubObject, 'SubmissionFile'));
-			$submissionDao = Application::getSubmissionDAO();
 			$submission = $submissionDao->getById($pubObject->getSubmissionId(), null, true);
+
 			if (!$submission) return null;
 			// Now we can identify the context.
 			$contextId = $submission->getJournalId();
 		}
+		// Get the latest submission revision
+		$latestRevision = $submissionDao->getLatestRevisionId($submission->getId());
```

Comment:
hmmm, the function getCurrentVersionId also uses $submissionDao->getLatestRevisionId. So there is no advantage here ... 🤔